site/src/pages/LoginPage/LoginPage.tsx

```diff
@@ -41,42 +49,19 @@ export const LoginPage: FC = () => {
 		});
 	}, [isSignedIn, buildInfoQuery.data, user?.id]);
 
-	if (isSignedIn) {
-		if (buildInfoQuery.data) {
-			// This uses `navigator.sendBeacon`, so window.href
-			// will not stop the request from being sent!
-			sendDeploymentEvent(buildInfoQuery.data, {
-				type: "deployment_login",
-				user_id: user?.id,
-			});
-		}
-
-		// If the redirect is going to a workspace application, and we
-		// are missing authentication, then we need to change the href location
-		// to trigger a HTTP request. This allows the BE to generate the auth
-		// cookie required.  Similarly for the OAuth2 exchange as the authorization
-		// page is served by the backend.
-		// If no redirect is present, then ignore this branched logic.
-		if (redirectTo !== "" && redirectTo !== "/") {
-			try {
-				// This catches any absolute redirects. Relative redirects
-				// will fail the try/catch. Subdomain apps are absolute redirects.
-				const redirectURL = new URL(redirectTo);
-				if (redirectURL.host !== window.location.host) {
-					window.location.href = redirectTo;
-					return null;
-				}
-			} catch {
-				// Do nothing
-			}
-			// Path based apps and OAuth2.
-			if (redirectTo.includes("/apps/") || redirectTo.includes("/oauth2/")) {
-				window.location.href = redirectTo;
-				return null;
-			}
-		}
-
-		return <Navigate to={redirectTo} replace />;
+	// The reason we need `window.location.href` for api redirects is that we
+	// need the page to reload and make a request to the backend. If we use
+	// `<Navigate>` react would handle the redirect itself and never request the
+	// page from the backend.
+	if (isSignedIn && isApiRouteRedirect) {
+		const sanitizedUrl = new URL(redirectTo, window.location.origin);
+		window.location.href = sanitizedUrl.pathname + sanitizedUrl.search;
+		return null;
+	}
+	if (isSignedIn && !isApiRouteRedirect) {
+		return (
+			<Navigate to={redirectUrl ? redirectUrl.pathname : redirectTo} replace />
```

Comment:
So I poked around a bit more of the code, and I'm just making sure: there's a second file that calls the `retrieveRedirect` function (`LoginPageView.tsx`). Do we need to update anything around that?

Because right now, the control flow is set up so that the component:
1. Gets the `redirectTo` value from the search param's `redirect` param
2. Passes it to the `SignInForm`
3. Passes that to the `OAuthSignInForm`
4. As long as GitHub is enabled as an auth method, a button shows up with an href that uses the raw `redirectTo` as a param

Not sure how the callback logic works, but would the updates to this file catch any issues from there?

Reply:
Yeah, good find. This is being sent directly to the backend which handles approving the redirect URLs, so it doesn't need any validation from the frontend.